### ISSUE TITLE
Default site without audio

### ISSUE THREAD

**Issue by glueckself** [project-alice-assistant/ProjectAlice#71]
Hi!

After installing Alice, it tries to talk to me on a siteId called default, which I don't have
```
[23:04:46] [Dialogue] was asked to start a session on site default
[23:04:46] [Dialogue] session with id 'cc88aaf5-020b-4526-82a1-319380b0d5c5' was started on site default
[23:04:46] [Tts] was asked to say "Welcome to Project Alice! There seems to be no user defined yet, let's do this! Please, tell me your name."
[23:05:10] [Dialogue] session with id 'cc88aaf5-020b-4526-82a1-319380b0d5c5' was ended on site default. The session timed out because the Audio Server component did not respond in a timely manner. Please ensure that the snips audio server is started and running correctly.
```
Where can I change the site id that Alice uses on start?

Additionally, the site id is asked in the ProjectAliceInstaller script, but it's only used for the snips.toml config. I guess this should also be used in Alice at some point?

My setup is also probably causing problems, Alice is running on a virtual machine without sound input/output, my site is currently my laptop (called abakus). I could work around and rename it to default, but I would prefer to solve it in Alice.

**Comment by Psychokiller1888** [comment]
How's your /etc/snips.toml looking? Is the siteId set in there?

**Comment by glueckself** [comment]
Yes, it is, but its called like I entered it in the installer ("base"). This leads to the entry in snips.toml:
```
[snips-audio-server]
bind = "base@mqtt"
```
It is set by the following code in installer.sh:
```bash
read -e -p $'\e[33mIn what room are you going to place my main unit? \e[0m' -i 'default' siteId
siteId=${siteId/_/ /.}
# ...
sed -i -e 's/\# bind = "default@mqtt"/bind = "'${siteId}'@mqtt"/' /etc/snips.toml
```
However, those are the only three occurrences of siteId in the installer.sh script.
Entering here anything but "default" leads to the problem that Alice can't connect to the audio server on the main unit. This might become a problem for other users.

In my case, the device with audio support has the siteId "abakus" (`bind = "abakus@mqtt"`). I could change that to `bind = "default@mqtt"`, but I'd prefer to change it inside Alice.

**Comment by glueckself** [comment]
Is there a line inside Alice or AliceCore where I can change the siteId from default?
Or should I work around it for now and rename the site on my laptop?

**Comment by Psychokiller1888** [comment]
It should be correctly supported, but there seems to be a bug, i'll control that in a while

**Comment by Psychokiller1888** [comment]
Hmmm, ConfigManager line 65 should load your site id and replace the 'default' by the name

I just extended it to
`constants.DEFAULT_SITE_ID = self._snipsConfigurations.get('snips-audio-server', {'bind': 'default@mqtt'}).get('bind', 'default@mqtt').replace('@mqtt', '')`

**Comment by glueckself** [comment]
Ah, thank you! That's what I've been looking for.

Does it make sense to put it in config.py, something like:
`"defaultSiteId": "!snips.toml"` for reading it from snips.toml and
`"defaultSiteId": "abakus"` for setting it explicitly.
It wouldn't need to be configured via the installer.

Because if I understand correctly, having it in snips.toml also configures the snips-audio-server to use it, which can't do anything on a device with no sound support.

Or is that just unnecessary complexity and my use case is too far from the intended use cases?
If that's so, then I suggest to mark this issue as resolved/won't fix.

**Comment by Psychokiller1888** [comment]
I think that adds a layer of complexity as then the info is both in snips.toml and in the configs. As of now it's directly taken from the toml file

**Comment by glueckself** [comment]
I'll change the bind in snips.toml to abakus@mqtt (my laptops site) and after starting Alice I'm going to stop the snips-audio-server service manually. Otherwise I would have two snips-audio-servers bound to abakus@mqtt (the real one on my laptop, the wrong one on my server).

I think that's it for this issue. Thank you for your support! :)

**Comment by glueckself** [comment]
I have to reopen the issue, because setting the bind value to "abakus@mqtt" in snips.toml didn't help.

The initial Alice interaction (Welcome to Project Alice ...) still went to the "default" site.
Only after I changed the DEFAULT_SITE_ID in core/commons/constants.py it used the right site.

This is a good enough workaround for me, but I think anyone changing the default value "default" when installing via the installer is going to have the same problem.

Also, this seems to be a problem only for the initial interaction. I was able to test the MpdClient module without any problems. I only see in snips-watch that Alice tries to say welcome via "default".
